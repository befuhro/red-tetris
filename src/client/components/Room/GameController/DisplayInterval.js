import React from 'react';

export default ({intervalMove}) => (<div><p  style={{color: 'white'}}>{intervalMove}</p></div>)
