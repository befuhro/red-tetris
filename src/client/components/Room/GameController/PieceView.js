import React from 'react';

import Square from '../Square';

const pieceViewStyle = {
    width: '11.95vh',
    height: '11.95vh',
    margin: '2.5vh',
    display: 'flex',
    flexWrap: 'wrap',
    borderColor: '#0f1423',
    borderWidth: '2vh 2vh 2vh 2vh',
    borderRadius: '1vh 1vh 1vh 1vh',
    borderStyle: 'inset'
};

function createArray(piece) {
    let array = Array(16).fill({color: 'white'});
    if (piece !== undefined) {
        piece.position.forEach((position) => {
            array[(position.row * 4) + position.column - 3] = { color: piece.color };
        });
    }
    return (array);
}

function PieceView({ piece }) {
    const array = createArray(piece);

    return (<div style={pieceViewStyle}>
        {
            array.map((square, index) =>
                <Square
                    key={index}
                    index={index}
                    color={square.color}
                />)
        }
    </div>);
}

// PieceView.propTypes = {
//     piece: PropTypes.object.isRequired,
// };

export default PieceView;
