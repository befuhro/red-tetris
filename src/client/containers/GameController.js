import { connect } from 'react-redux';
import GameController from '../components/GameController.js';
import { startParty} from "../actions/game";

const mapStateToProps = (state) => ({
    isRoomLeader: state.env.isRoomLeader,
    piece0: state.game.pieces[0],
    piece1: state.game.pieces[1],
    piece2: state.game.pieces[2],
    gameIsStarted: state.game.isStarted
});

const mapDispatchToProps = (dispatch) => ({
    // startParty: () => startParty(dispatch, ownProps),
    startParty: () => dispatch(startParty()),
});


export default connect(mapStateToProps, mapDispatchToProps)(GameController);