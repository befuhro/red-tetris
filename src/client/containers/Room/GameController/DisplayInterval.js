import {connect} from 'react-redux';

import DisplayInterval from '../../../components/Room/GameController/DisplayInterval';

const mapStateToProps = (state) => ({intervalMove: state.room.intervalMove,});

export default connect(mapStateToProps)(DisplayInterval);
