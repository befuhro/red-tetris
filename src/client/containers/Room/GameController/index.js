import { connect } from 'react-redux';

import GameController from '../../../components/Room/GameController';

const mapStateToProps = (state) => ({
    isRoomLeader: state.room.isRoomLeader,
    firstPiece: state.room.pieces[0],
    secondPiece: state.room.pieces[1],
    thirdPiece: state.room.pieces[2],
    gameIsStarted: state.room.gameIsStarted,
    socket: state.room.socket
});

const mapDispatchToProps = dispatch => ({ dispatch: dispatch });

export default connect(mapStateToProps, mapDispatchToProps)(GameController);
