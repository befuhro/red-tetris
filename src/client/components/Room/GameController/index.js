import React from 'react';

import PieceView from './PieceView';
import { startParty } from "../../../actions/room/roomActions";

const gameControllerStyle = {
    justifySelf: 'end',
    width: '22vh',
    height: '70vh',
    margin: '12.8vh 0 0 0',
    padding: '2vh 2vh 0 2vh',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    backgroundColor: '#0b1d2d',
    borderColor: '#0b1d2d',
    borderWidth: '1vh 2vh 0 2vh',
    borderStyle: 'outset',
};

const buttonStyle = {
    padding: '1vh',
    fontFamily: 'Orbitron',
    fontWeight: '500',
    fontSize: '2vh',
    textAlign: 'center',
    color: '#e1301d',
    backgroundColor: '#5a6978',
    borderColor: '#0f1423',
    borderWidth: '1vh 1vh 1vh 1vh',
    borderRadius: '1vh 1vh 1vh 1vh',
    borderStyle: 'outset'
};

function StartButton({ isRoomLeader, gameIsStarted, dispatch }) {
    return (
        <div>
            {isRoomLeader && gameIsStarted !== true ?
                <button style={buttonStyle} onClick={() => dispatch(startParty())}>Start Game</button>
                :
                null
            }
        </div>
    );
}

export default ({ isRoomLeader, gameIsStarted, firstPiece, secondPiece, thirdPiece, dispatch}) => {
    return (
        <div style={gameControllerStyle}>
            <div>
                <PieceView piece={thirdPiece}/>
                <PieceView piece={secondPiece}/>
                <PieceView piece={firstPiece}/>
            </div>
            <StartButton isRoomLeader={isRoomLeader} gameIsStarted={gameIsStarted} dispatch={ dispatch }/>
        </div>);
}
