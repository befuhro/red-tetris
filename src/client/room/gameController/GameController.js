import React from 'react';
import { connect } from 'react-redux';
// import PropTypes from 'prop-types';

import { startParty} from "../../Store/Actions/Game";
import PieceView from './PieceView';

const gameControllerStyle = {
    justifySelf: 'end',
    width: '22vh',
    height: '70vh',
    margin: '12.8vh 0 0 0',
    padding: '2vh 2vh 0 2vh',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    backgroundColor: '#0b1d2d',
    borderColor: '#0b1d2d',
    borderWidth: '1vh 2vh 0 2vh',
    borderStyle: 'outset',
};

const buttonStyle = {
    padding: '1vh',
    fontFamily: 'Orbitron',
    fontWeight: '500',
    fontSize: '2vh',
    textAlign: 'center',
    color: '#e1301d',
    backgroundColor: '#5a6978',
    borderColor: '#0f1423',
    borderWidth: '1vh 1vh 1vh 1vh',
    borderRadius: '1vh 1vh 1vh 1vh',
    borderStyle: 'outset'

};

function StartButton({ isRoomLeader, gameIsStarted, startParty }) {
    return (
        <div>
            {isRoomLeader && gameIsStarted === false ?
                <button style={buttonStyle} onClick={startParty}>Start Game</button>
                :
                null
            }
        </div>
    );
}

function GameController({ isRoomLeader, piece0, piece1, piece2, gameIsStarted, startParty }) {
    console.log(gameIsStarted);
    return (
        <div style={gameControllerStyle}>
            <div>
                <PieceView piece={piece2}/>
                <PieceView piece={piece1}/>
                <PieceView piece={piece0}/>
            </div>
            <StartButton isRoomLeader={isRoomLeader} gameIsStarted={gameIsStarted} startParty={startParty}/>
        </div>);
}

// GameController.propTypes = {
//     onLoad: PropTypes.func.isRequired,
//     isListening: PropTypes.bool.isRequired,
// };

const mapStateToProps = (state) => ({
    isRoomLeader: state.env.isRoomLeader,
    piece0: state.game.pieces[0],
    piece1: state.game.pieces[1],
    piece2: state.game.pieces[2],
    gameIsStarted: state.game.isStarted
});

const mapDispatchToProps = (dispatch) => ({
    // startParty: () => startParty(dispatch, ownProps),
    startParty: () => dispatch(startParty()),
});

export default connect(mapStateToProps, mapDispatchToProps)(GameController);