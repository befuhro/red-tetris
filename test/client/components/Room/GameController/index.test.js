import * as React from 'react';
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

enzyme.configure({adapter: new Adapter()});

import GameController from '../../../../../src/client/components/Room/GameController';

describe('<GameController>', () => {
    describe('test leader room and game not started', () => {
        let component;

        beforeAll(() => {
            component = enzyme.mount(
                <GameController
                    isRoomLeader={true}
                    gameIsStarted={false}
                    firstPiece={{
                        position: [{column: 4, row: 0}, {column: 4, row: 1}, {column: 4, row: 2}, {column: 4, row: 3}],
                        indexRotation: 1,
                        rotation: [
                            {column: -1, row: 1}, {column: 0, row: 0}, {column: 1, row: -1}, {column: 2, row: -2},
                            {column: 2, row: -1}, {column: 1, row: 0}, {column: 0, row: 1}, {column: -1, row: 2},
                            {column: 1, row: 2}, {column: 0, row: 1}, {column: -1, row: 0}, {column: -2, row: -1},
                            {column: -2, row: -2}, {column: -1, row: -1}, {column: 0, row: 0}, {column: 1, row: 1},
                        ],
                        color: 'lightblue',
                    }}
                    secondPiece={{
                        position: [{column: 4, row: 0}, {column: 4, row: 1}, {column: 4, row: 2}, {column: 4, row: 3}],
                        indexRotation: 1,
                        rotation: [
                            {column: -1, row: 1}, {column: 0, row: 0}, {column: 1, row: -1}, {column: 2, row: -2},
                            {column: 2, row: -1}, {column: 1, row: 0}, {column: 0, row: 1}, {column: -1, row: 2},
                            {column: 1, row: 2}, {column: 0, row: 1}, {column: -1, row: 0}, {column: -2, row: -1},
                            {column: -2, row: -2}, {column: -1, row: -1}, {column: 0, row: 0}, {column: 1, row: 1},
                        ],
                        color: 'lightblue',
                    }}
                    thirdPiece={{
                        position: [{column: 4, row: 0}, {column: 4, row: 1}, {column: 4, row: 2}, {column: 4, row: 3}],
                        indexRotation: 1,
                        rotation: [
                            {column: -1, row: 1}, {column: 0, row: 0}, {column: 1, row: -1}, {column: 2, row: -2},
                            {column: 2, row: -1}, {column: 1, row: 0}, {column: 0, row: 1}, {column: -1, row: 2},
                            {column: 1, row: 2}, {column: 0, row: 1}, {column: -1, row: 0}, {column: -2, row: -1},
                            {column: -2, row: -2}, {column: -1, row: -1}, {column: 0, row: 0}, {column: 1, row: 1},
                        ],
                        color: 'lightblue',
                    }}
                    socket={null}
                    dispatch={null}
                />
            );
        });
        it('should return basic GameController', () => expect(component.html()).toMatchSnapshot());
    });



    describe('test not leader room and game started', () => {
        let component;

        beforeAll(() => {
            component = enzyme.mount(
                <GameController
                    isRoomLeader={false}
                    gameIsStarted={true}
                    firstPiece={{
                        position: [{column: 4, row: 0}, {column: 4, row: 1}, {column: 4, row: 2}, {column: 4, row: 3}],
                        indexRotation: 1,
                        rotation: [
                            {column: -1, row: 1}, {column: 0, row: 0}, {column: 1, row: -1}, {column: 2, row: -2},
                            {column: 2, row: -1}, {column: 1, row: 0}, {column: 0, row: 1}, {column: -1, row: 2},
                            {column: 1, row: 2}, {column: 0, row: 1}, {column: -1, row: 0}, {column: -2, row: -1},
                            {column: -2, row: -2}, {column: -1, row: -1}, {column: 0, row: 0}, {column: 1, row: 1},
                        ],
                        color: 'lightblue',
                    }}
                    secondPiece={{
                        position: [{column: 4, row: 0}, {column: 4, row: 1}, {column: 4, row: 2}, {column: 4, row: 3}],
                        indexRotation: 1,
                        rotation: [
                            {column: -1, row: 1}, {column: 0, row: 0}, {column: 1, row: -1}, {column: 2, row: -2},
                            {column: 2, row: -1}, {column: 1, row: 0}, {column: 0, row: 1}, {column: -1, row: 2},
                            {column: 1, row: 2}, {column: 0, row: 1}, {column: -1, row: 0}, {column: -2, row: -1},
                            {column: -2, row: -2}, {column: -1, row: -1}, {column: 0, row: 0}, {column: 1, row: 1},
                        ],
                        color: 'lightblue',
                    }}
                    thirdPiece={{
                        position: [{column: 4, row: 0}, {column: 4, row: 1}, {column: 4, row: 2}, {column: 4, row: 3}],
                        indexRotation: 1,
                        rotation: [
                            {column: -1, row: 1}, {column: 0, row: 0}, {column: 1, row: -1}, {column: 2, row: -2},
                            {column: 2, row: -1}, {column: 1, row: 0}, {column: 0, row: 1}, {column: -1, row: 2},
                            {column: 1, row: 2}, {column: 0, row: 1}, {column: -1, row: 0}, {column: -2, row: -1},
                            {column: -2, row: -2}, {column: -1, row: -1}, {column: 0, row: 0}, {column: 1, row: 1},
                        ],
                        color: 'lightblue',
                    }}
                    socket={null}
                    dispatch={null}
                />
            );
        });
        it('should return basic GameController', () => expect(component.html()).toMatchSnapshot());
    });
});
