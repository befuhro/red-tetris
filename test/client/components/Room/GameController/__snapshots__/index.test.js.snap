// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameController> test leader room and game not started should return basic GameController 1`] = `"<div style=\\"justify-self: end; width: 22vh; height: 70vh; margin: 12.8vh 0px 0px 0px; padding: 2vh 2vh 0px 2vh; display: flex; flex-direction: column; align-items: center; background-color: rgb(11, 29, 45); border-color: #0b1d2d; border-width: 1vh 2vh 0px 2vh; border-style: outset;\\"><div><div style=\\"width: 6.8vh; height: 6.8vh; margin: 2.5vh; display: flex; flex-wrap: wrap; border-color: #0f1423; border-width: 2vh 2vh 2vh 2vh; border-radius: 1vh 1vh 1vh 1vh; border-style: inset;\\"><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(22, 227, 220);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(22, 227, 220);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(22, 227, 220);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(22, 227, 220);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div></div><div style=\\"width: 6.8vh; height: 6.8vh; margin: 2.5vh; display: flex; flex-wrap: wrap; border-color: #0f1423; border-width: 2vh 2vh 2vh 2vh; border-radius: 1vh 1vh 1vh 1vh; border-style: inset;\\"><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(22, 227, 220);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(22, 227, 220);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(22, 227, 220);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(22, 227, 220);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div></div><div style=\\"width: 6.8vh; height: 6.8vh; margin: 2.5vh; display: flex; flex-wrap: wrap; border-color: #0f1423; border-width: 2vh 2vh 2vh 2vh; border-radius: 1vh 1vh 1vh 1vh; border-style: inset;\\"><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(22, 227, 220);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(22, 227, 220);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(22, 227, 220);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(22, 227, 220);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div></div></div><div><button style=\\"padding: 1vh; font-family: Orbitron; font-weight: 500; font-size: 2vh; text-align: center; color: rgb(225, 48, 29); background-color: rgb(90, 105, 120); border-color: #0f1423; border-width: 1vh 1vh 1vh 1vh; border-radius: 1vh 1vh 1vh 1vh; border-style: outset;\\">Start Game</button></div></div>"`;

exports[`<GameController> test not leader room and game started should return basic GameController 1`] = `"<div style=\\"justify-self: end; width: 22vh; height: 70vh; margin: 12.8vh 0px 0px 0px; padding: 2vh 2vh 0px 2vh; display: flex; flex-direction: column; align-items: center; background-color: rgb(11, 29, 45); border-color: #0b1d2d; border-width: 1vh 2vh 0px 2vh; border-style: outset;\\"><div><div style=\\"width: 6.8vh; height: 6.8vh; margin: 2.5vh; display: flex; flex-wrap: wrap; border-color: #0f1423; border-width: 2vh 2vh 2vh 2vh; border-radius: 1vh 1vh 1vh 1vh; border-style: inset;\\"><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(22, 227, 220);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(22, 227, 220);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(22, 227, 220);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(22, 227, 220);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div></div><div style=\\"width: 6.8vh; height: 6.8vh; margin: 2.5vh; display: flex; flex-wrap: wrap; border-color: #0f1423; border-width: 2vh 2vh 2vh 2vh; border-radius: 1vh 1vh 1vh 1vh; border-style: inset;\\"><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(22, 227, 220);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(22, 227, 220);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(22, 227, 220);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(22, 227, 220);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div></div><div style=\\"width: 6.8vh; height: 6.8vh; margin: 2.5vh; display: flex; flex-wrap: wrap; border-color: #0f1423; border-width: 2vh 2vh 2vh 2vh; border-radius: 1vh 1vh 1vh 1vh; border-style: inset;\\"><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(22, 227, 220);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(22, 227, 220);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(22, 227, 220);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(22, 227, 220);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div><div style=\\"height: 1.5vh; width: 1.5vh; border-style: outset; border-width: 0.1vh; border-color: #749099; background-color: rgb(90, 105, 120);\\"> </div></div></div><div></div></div>"`;
